client/navigation/components/container/test/index.js

```diff
@@ -0,0 +1,252 @@
+		parent: 'woocommerce',
+		backButtonLabel: 'Custom Back Label',
+		menuId: 'primary',
+	},
+	{
+		id: 'primary-child',
+		title: 'Primary Child',
+		parent: 'primary-category',
+		menuId: 'primary',
+		url: 'admin.php?page=wc-admin&path=/child',
+	},
+	{
+		id: 'favorite-category',
+		title: 'Favorite Category',
+		isCategory: true,
+		parent: 'woocommerce',
+		menuId: 'favorites',
+	},
+	{
+		id: 'favorite-child',
+		title: 'Favorite Child',
+		parent: 'favorite-category',
+		menuId: 'plugins',
+	},
+	{
+		id: 'plugin-category',
+		title: 'Plugin Category',
+		isCategory: true,
+		parent: 'woocommerce',
+		menuId: 'plugins',
+	},
+	{
+		id: 'plugin-child',
+		title: 'Plugin Child',
+		parent: 'plugin-category',
+		menuId: 'plugins',
+		url: 'admin.php?page=my-plugin',
+	},
+	{
+		id: 'secondary-category',
+		title: 'Secondary Category',
+		isCategory: true,
+		parent: 'woocommerce',
+		menuId: 'secondary',
+	},
+	{
+		id: 'secondary-child',
+		title: 'Secondary Child',
+		parent: 'secondary-category',
+		menuId: 'secondary',
+	},
+];
+
+describe( 'Container', () => {
+	beforeEach( () => {
+		delete window.location;
+		window.location = new URL( getAdminLink( 'admin.php?page=wc-admin' ) );
+
+		useSelect.mockImplementation( () => ( {
+			menuItems,
+			favorites: [ 'favorite-category' ],
+		} ) );
+	} );
+
+	afterAll( () => {
+		window.location = originalLocation;
+	} );
+
+	test( 'should show the woocommerce root items initially', () => {
+		const { container } = render( <Container /> );
+
+		expect(
+			container.querySelectorAll( '.components-navigation__item' ).length
+		).toBe( 5 );
+	} );
+
+	test( 'should set the active item based on initial location', async () => {
+		const { queryByText } = render( <Container /> );
+
+		expect(
+			queryByText( 'Home' ).parentElement.parentElement.classList
+		).toContain( 'is-active' );
+	} );
+
+	test( 'should set the initial active item based on current location', () => {
+		window.location = new URL( getAdminLink( 'admin.php?page=my-plugin' ) );
+
+		const { container, queryByText } = render( <Container /> );
+
+		expect(
+			container.querySelector( '.woocommerce-navigation-category-title' )
+				.textContent
+		).toBe( 'Plugin Category' );
+		expect(
+			queryByText( 'Plugin Child' ).parentElement.parentElement.classList
+		).toContain( 'is-active' );
+	} );
+
+	test( 'should update the active item and level when location changes', async () => {
+		window.location = new URL( getAdminLink( 'admin.php?page=wc-admin' ) );
+
+		const { container, queryByText } = render( <Container /> );
+
+		expect( queryByText( 'Primary Child' ) ).toBeNull();
+
+		// Trigger and wait for history change.
+		await act( async () => {
+			delete window.location;
+			window.location = new URL(
+				getAdminLink( 'admin.php?page=wc-admin&path=/child' )
+			);
+			getHistory().push( getAdminLink( '/child' ) );
+
+			await new Promise( ( resolve ) => {
+				setTimeout( () => {
+					resolve();
+				}, 0 );
+			} );
```

Comment:
Is this to wait for the event loop to finish?

Reply:
Exactly.  We have a similar check in the history listener to correctly set the active item, so it also needs to be done here.